app/models/pacbio/library.rb

```diff
@@ -86,5 +92,22 @@ def check_for_derived_aliquots
       errors.add(:base, 'Cannot delete a library that is used in a pool or run')
       throw(:abort)
     end
+
+    # Checks if the volume has been changed and if the new volume is greater than the used volume.
+    # If the volume has not been changed, the method immediately returns.
+    # If the new volume is greater than the used volume, the method returns true.
+    # If the new volume is less than or equal to the used volume
+    # - an error is added to the volume attribute and
+    # - the update operation is aborted.
+    #
+    # @return [TrueClass, nil]
+    # Returns true if the new volume is greater than the used volume, nil otherwise.
+    def check_volume
+      return unless volume_changed?
+      return true if volume > used_volume
```

Comment:
I don't know whether we want to be checking the volume or the primary_aliquot volume. Currently the request from the UI will include and update both the library volume and primary_aliquot volume.

Reply:
I think we should check the primary_aliquot volume as this is what we will be using in the future. I presume we will eventually deprecate volume?

Reply:
The method changed to checking the primary aliquot volume